Improve vocab delete

l5r opened the pull request:
This PR improves vocab delete feedback according to https://github.com/redpencilio/app-vocabsearch/pull/17

- Delete confirmation dialog closes immediately
- Vocab list shows deletion status (deleting/waiting for mu-search)
- Vocab edit page gets blocked, informing the user of deletion

l5r committed:
Move vocab delete to background task

Deleting a vocab now happens in the background, and the vocab list UI
shows the status of the delete task.

l5r committed:
Wait for vocabs to be completely deleted

Now we also wait for the vocab concepts to be removed from muu-search,
and provide feedback to the user.

x-m-el commented on app/templates/vocabularies/index.hbs:
```diff
@@ -21,30 +21,44 @@
+          {{#if deleting}}
+            Deleting <TaskStatus @task={{await vocab.deletingTask}} @onTaskFinished={{action 'refresh'}} />
+          {{else if waiting}}
+            Waiting for search <TaskStatus @task={{await vocab.deleteWaitingTask}} @onTaskFinished={{action 'refresh'}} />
```
`action` helper is deprecated
Just call `this.refresh` instead

x-m-el commented on app/models/vocabulary.js:
```diff
@@ -1,11 +1,49 @@
+      'page[size]': 1
+    })
+    return tasks.firstObject
+  }
```
Understandable to do this here, but in Ember it is a bad idea to put any kind of logic like this on the model.
On modern ember this won't even be possible anymore, so this would be deprecated.

Only add simpel helpers that are derived from the data (like is done for task model) or when needing to override something (like the `save` function)

x-m-el reviewed (changes requested):
There is better feedback during the deletion, but I don't think it shows any information if a deletion failed/timed out.

l5r committed:
Remove deprecated action helpers

l5r committed:
Move deleting task out of model in to helper

In Ember it is a bad idea to put any kind of logic like this on the
model. On modern ember this won't even be possible anymore, so this
would be deprecated.

x-m-el reviewed (changes requested):
the text "Note that deleting large vocabularies may take a long time. Make sure to refresh the page manually." can now be changed/removed, because there is no manual refresh needed anymore.

task-status component gives errors because of task being null. Might be fixed adding `{{#if @task}}` to the template, depending on the reason for the error.

l5r committed:
Remove page refresh message

Remove message telling the user to refresh the page.

l5r committed:
Fix tasks being 'null'

The await helper temporarily returns null until the promise resolves.

x-m-el commented:
When deleting and it failing (e.g. waiting for search times out), it doesn't show an error occurred or any other feedback.
It should probably show something to indicate a previous job failed.

l5r committed:
Show failed delete tasks

Interface would refresh, even when a task was failed.

x-m-el commented:
When a delete task doesn't start for some reason, it seems to be stuck indefinitely. Not sure why the task scheduler doesn't pick it up. Maybe it only runs tasks that come specifically from a delta only.
I don't have a reproduction for this bug yet.
Either way, I think that reshowing the delete button if a task hasn't started after some time might be a fix, if not too hard to do. In some cases that might trigger the delete again correctly.
That's what is possible for the other pipeline buttons (like downloading file): even if the task is still busy, it is possible to click it again to restart it, which can help if the task somehow got stuck.

The same problem occurs when mu-search wait fails: it keeps showing the it failed, but no way to restart the delete somehow.

x-m-el commented:
PR moved to https://github.com/redpencilio/frontend-vocabserver/pull/30 (PR from non-forked branch)
